R/land_leaf.R

```diff
@@ -157,16 +157,20 @@ LandLeaf_getCalLandAllocation <- function(aLandLeaf, aPeriod) {
 #' @param aChoiceFnAbove Type of logit
 #' @param aPeriod Model time period
 #' @param aParent Parent node (needed for calibrating new techs)
+#' @param aScenarioInfo Scenario info object
 #' @author KVC September 2017
-LandLeaf_calculateShareWeights <- function(aLandLeaf, aChoiceFnAbove, aPeriod, aParent) {
-  # TODO: move output cost to a member variable; implement absolute cost logit
-  if( aChoiceFnAbove$mType == "relative-cost") {
-    aLandLeaf$mShareWeight[aPeriod] <- RelativeCostLogit_calcShareWeight(aChoiceFnAbove,
-                                                              aLandLeaf$mShare[[aPeriod]],
-                                                              aLandLeaf$mProfitRate[[aPeriod]],
-                                                              aPeriod)
-  } else {
-    print("ERROR: Invalid choice function in LandLeaf_calculateShareWeight")
+LandLeaf_calculateShareWeights <- function(aLandLeaf, aChoiceFnAbove, aPeriod, aParent, aScenarioInfo) {
+  # We only want to calculate share weights if this scenario requires it. Otherwise, we use read in values
+  if(aScenarioInfo$mCalibrateShareWt == TRUE) {
+    # TODO: move output cost to a member variable; implement absolute cost logit
+    if( aChoiceFnAbove$mType == "relative-cost") {
+      aLandLeaf$mShareWeight[aPeriod] <- RelativeCostLogit_calcShareWeight(aChoiceFnAbove,
```

Comment:
That is an interesting question. We put in the indexing by period for biomass, which is the only land type with a share weight that changes over time (going from 0 in the historical period to a positive number in future periods). Since the read in values are only applicable for the other land types, I didn't bother putting in a time vector. I think the current approach is consistent with current GCAM, but we may want to discuss this further at some point.